hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -181,7 +171,7 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
     HoodieCommitMetadata metadata = CommitUtils.buildMetadata(stats, partitionToReplaceFileIds, extraMetadata, operationType, config.getSchema(), commitActionType);
     // Finalize write
     finalizeWrite(table, instantTime, stats);
-
+    HeartbeatUtils.abortIfHeartbeatExpired(instantTime, table, heartbeatClient, config);
```

Comment:
I want to avoid business logic like approach 2 in the `AbstractHoodieWriteClient`, this logic is pushed to the method `abortIfHeartbeatExpired`. Alternatively, the method `abortIfHeartbeatExpired` build on other public API's to be able to do custom business logic if needed.